### PR TITLE
[BugFix] Show resource group with empty classifiers

### LINKED ISSUE #15930: Couldn't create resource group
### Steps to reproduce the behavior (Required)

1. create the resource group
```
CREATE RESOURCE GROUP `testwg`TO (query_type in ('select')) WITH (
    'type' = 'normal',
    'cpu_core_limit' = '20',
    'mem_limit' = '90%',
    'concurrency_limit' = '98'
);
```

2. show the resource group to found the classifiers id
```
MySQL [(none)]> show resource group testwg\G
*************************** 1. row ***************************
                      name: testwg
                        id: 959450
            cpu_core_limit: 20
                 mem_limit: 90.0%
big_query_cpu_second_limit: 0
 big_query_scan_rows_limit: 0
       big_query_mem_limit: 0
         concurrency_limit: 98
                      type: NORMAL
               classifiers: (id=959451, weight=1.1, query_type in (SELECT))
1 row in set (0.00 sec)
```

3.  drop the classifiers of resource group
```
MySQL [(none)]> ALTER RESOURCE GROUP testwg DROP (959451);
Query OK, 0 rows affected (0.00 sec)
```

4. show the resource group again report empty
```
MySQL [(none)]> show resource group testwg\G
Empty set (0.00 sec)
```

5. create the esource group after drop report exists
```
MySQL [(none)]> CREATE RESOURCE GROUP `testwg`TO (query_type in ('select')) WITH (
    ->     'type' = 'normal',
    ->     'cpu_core_limit' = '20',
    ->     'mem_limit' = '90%',
    ->     'concurrency_limit' = '98'
    -> );
ERROR 1064 (HY000): RESOURCE_GROUP(testwg) already exists
```

### Expected behavior (Required)

foriden to drop the last classifiers of resource group

### Real behavior (Required)

drop last classifiers of resource group successfully and show empty but could not reate again

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3.4

### PR DESCRIPTION
**Pull request by mchades**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15930 

## Problem Summary(Required) ：

show resource group with empty classifiers for reuse

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ResourceGroup.java|3|3|100.00%|[]|

**Review by imay — DISMISSED** [review]
LGTM

**Comment by imay** [comment]
run starrocks_admit_test

**Comment by imay** [comment]
@mchades seems some admit tests failed. Could take a look when you are free.

**Comment by colorfulu** [comment]
run starrocks_admit_test

**Comment by colorfulu** [comment]
run starrocks_admit_test

File: fe/fe-core/src/test/java/com/starrocks/analysis/ResourceGroupStmtTest.java [hunk 1]
```diff
@@ -314,7 +316,11 @@ public void testAlterResourceGroupDropManyClassifiers() throws Exception {
         String rgName = rows.get(0).get(0);
         List<String> classifierIds = rows.stream().filter(row -> row.get(0).equals(rgName))
                 .map(row -> getClassifierId(row.get(row.size() - 1))).collect(Collectors.toList());
-        rows = rows.stream().filter(row -> !row.get(0).equals(rgName)).collect(Collectors.toList());
+        rows = rows.stream().peek(row -> {
+            if (row.get(0).equals(rgName)) {
+                row.set(9, "(weight=0.0)");
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ResourceGroupStmtTest.java:321]: Please do not use this magic number.

**Comment by colorfulu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ResourceGroup.java|3|3|100.00%|[]|

**Comment by mchades** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**